intellij/src/saros/intellij/ui/util/UIProjectUtils.java

```diff
@@ -0,0 +1,105 @@
+package saros.intellij.ui.util;
+
+import com.intellij.ide.DataManager;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.actionSystem.DataKeys;
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.concurrency.Promise;
+import saros.session.ISarosSession;
+import saros.session.ISarosSessionManager;
+import saros.session.ISessionLifecycleListener;
+import saros.session.SessionEndReason;
+
+/**
+ * Class to provide access to an active project object. This class should only be used for UI
+ * purposes. To interact with the project whose module is shared as part of a Saros session, please
+ * use the Project object contained in the session context.
+ */
+public class UIProjectUtils {
+  private volatile ISarosSession sarosSession;
+
+  @SuppressWarnings("FieldCanBeLocal")
+  private ISessionLifecycleListener sessionLifecycleListener =
+      new ISessionLifecycleListener() {
+        @Override
+        public void sessionStarted(ISarosSession session) {
+          sarosSession = session;
+        }
+
+        @Override
+        public void sessionEnded(ISarosSession session, SessionEndReason reason) {
+          sarosSession = null;
+        }
+      };
+
+  public UIProjectUtils(ISarosSessionManager sarosSessionManager) {
+    sarosSessionManager.addSessionLifecycleListener(sessionLifecycleListener);
+  }
+
+  /**
+   * Returns the <code>Project</code> object contained in the session context or <code>null</code>
+   * if there currently is no session.
+   *
+   * <p><b>NOTE:</b> This method should only be used to for UI purposes. To interact with the
+   * project whose module is shared as part of a Saros session, please use the Project object
+   * contained in the session context.
+   *
+   * @return the <code>Project</code> object contained in the session context or <code>null</code>
+   *     if there currently is no session.
+   */
+  @Nullable
+  private Project getProjectFromSession() {
+    ISarosSession currentSession = sarosSession;
+
+    if (currentSession == null) {
+      return null;
+    }
+
+    return sarosSession.getComponent(Project.class);
```

Comment:
Use `currentSession` object.

Reply:
I use the additional variable to ensure that I always work on the same object. The variable `sarosSession` is volatile and can be changed asynchronously through the ISessionLifecycleListener defined in the beginning of the class when the session state changes (session starts/ends). If I would work on the volatile reference directly, it could happen that the session ends after my check but before I am working with the variable, which would lead to a NPE as the variable would have been set to `NULL` by the session listener in accordance with the session end.